# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeParent.h
# Commit: 9bfd4e60ec4e
# Full Hash: 9bfd4e60ec4e36b680c1722a3bd3e4b7359f7e36
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Always send parent commands when sending mDestroyedActors r=kats,sotaro
#   
#   If we try to send them separately as we were before, we can run into
#   cases where we try to destroy the actors and then send the OpRemoveTexture,
#   which crashes.
# ==============================================================================

diff -r 8761ce294d2c -r 9bfd4e60ec4e gfx/layers/wr/WebRenderBridgeParent.h
--- a/gfx/layers/wr/WebRenderBridgeParent.h	Fri Mar 22 18:29:00 2019 +0000
+++ b/gfx/layers/wr/WebRenderBridgeParent.h	Fri Mar 22 18:29:04 2019 +0000
@@ -296,17 +296,14 @@
   }
 
   bool SetDisplayList(wr::RenderRoot aRenderRoot, const gfx::IntRect& aRect,
-                      const nsTArray<WebRenderParentCommand>& aCommands,
                       const wr::LayoutSize& aContentSize, ipc::ByteBuf&& aDL,
                       const wr::BuiltDisplayListDescriptor& aDLDesc,
                       const nsTArray<OpUpdateResource>& aResourceUpdates,
                       const nsTArray<RefCountedShmem>& aSmallShmems,
                       const nsTArray<ipc::Shmem>& aLargeShmems,
                       const TimeStamp& aTxnStartTime,
-                      wr::TransactionBuilder& aTxn,
-                      Maybe<wr::AutoTransactionSender>& aTxnSender,
-                      wr::Epoch aWrEpoch, bool aValidTransaction,
-                      bool aObserveLayersUpdate);
+                      wr::TransactionBuilder& aTxn, wr::Epoch aWrEpoch,
+                      bool aValidTransaction, bool aObserveLayersUpdate);
 
   void UpdateAPZFocusState(const FocusTarget& aFocus);
   void UpdateAPZScrollData(const wr::Epoch& aEpoch, WebRenderScrollData&& aData,